src/coreclr/pal/src/init/pal.cpp

```diff
@@ -147,6 +143,42 @@ static bool RunningNatively()
 }
 #endif // __APPLE__
 
+#if defined(__sun)
+//
+// Let this program be "privilege aware". See privileges(7)
+//
+// Add the "proc_lock_memory" privilege needed for the later mlock in:
+// PAL_InitializeCoreCLR / InitializeFlushProcessWriteBuffers
+//
+// This privilege is added to the "inheritable" set via the exec_attr(5)
+// by adding a file like: /etc/security/exec_attr.d/dotnet  containing:
```

Comment:
Expecting user to create this entry feels like a similar workaround as asking them to run `sudo usermod -K defaultpriv='basic,proc_lock_memory' $(whoami)` (current workaround). I think we should find a better way to avoid calling mlock(), as we do for macOS and other membarrier() supported platforms.

We were discussing about fork et al. based solution, did it not work?

Reply:
I am looking at better options, but want to get through self hosting before I invest more time in that.
The exec_attr method is a bit easier for me to deal with for testing.

Reply:
Lets keep this patch in local branch. I don't think it is a good idea to trade one semi-finished solution with another of a kind. We had a long discussion here: https://github.com/dotnet/runtime/issues/36632.

Reply:
OK, let's just close this then.

Reply:
Alternatively, we can mark it as "draft" (there is a button visible for you on this page). In case we can't avoid mlock() this solution is better for packaging as you explained on IRC.